Documentation/commands.md

```diff
@@ -209,8 +209,37 @@ FOO=bar
 EXAMPLE_OVERRIDE=over
 ```
 
+#### Logging
+
+To get the logs you need to get pod's machine name. You can use machinectl
+
+```
+$ machinectl
+MACHINE                                  CLASS     SERVICE
+rkt-f241c969-1710-445a-8129-d3a7ffdd9a60 container nspawn
+
+1 machines listed.
+```
+
+or `rkt list --full`
+
+```
+# rkt list --full
+UUID                                    ACI     STATE   NETWORKS
+f241c969-1710-445a-8129-d3a7ffdd9a60    busybox running
+```
+
+Pod's machine name will be the pod's UUID with a `rkt-` prefix.
+
+Then you can use systemd's journalctl with `_HOSTNAME` set to the machine name:
+
+```
+$ journalctl -m _HOSTNAME=rkt-f241c969-1710-445a-8129-d3a7ffdd9a60
```

Comment:
We're using `--link-journal=try-host` to make this work with overlayfs.

There're two ways to link the journal:

* `--link-journal=[try]guest`: it creates a symlink from `/var/log/journal/[machine-id]` to container's `/container/dir/var/log/journal/[machine-id]`. This doesn't work for us because when we use overlayfs, we can't access container's filesystem.

* `--link-journal=[try]host`: It works bind-mounting `/var/log/journal/[machine-id]` from the host to `/container/dir/var/log/journal/[machine-id]`. That's what we use.

`journalctl -M` reads `/run/systemd/machines/[machine-name]`, gets its ROOT directory and tries to get the journal from `$ROOT/var/log/journal/[machine-id]`. This doesn't work with `--link-journal=[try]host` because `$ROOT/var/log/journal/[machine-id]` is a bind-mount and can't be read outside the mount namespace we create. (I have to investigate this more and maybe patch systemd to fix that).

TL;DR

We use `--link-journal=tryhost` and we use `journalctl -m` (it merges all the journal). Then we filter by machine. It's kind of a hack but it works.